src/Servers/IIS/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp

```diff
@@ -259,7 +265,7 @@ IN_PROCESS_APPLICATION::ExecuteApplication()
             if (m_waitForShutdown)
             {
                 const auto clrWaitResult = WaitForSingleObject(m_clrThread.native_handle(), m_pConfig->QueryShutdownTimeLimitInMS());
-                THROW_LAST_ERROR_IF(waitResult == WAIT_FAILED);
+                THROW_LAST_ERROR_IF(clrWaitResult == WAIT_FAILED);
```

Comment:
This is unrelated, I just noticed that we were checking the wrong wait timeout here.